### ISSUE TITLE
Localize spark.kerberos.keytab  from remote FileSystem

### ISSUE THREAD

**Issue by yaooqinn** [NetEase/kyuubi#672]

**Comment by wForget** [comment]
Is it possible to first support the keytab to be stored locally on the kyuubi server. 
like:
```
if(USE_LOCAL_KEYTAB) {
  conf.setIfMissing("spark.kerberos.principal", s"$proxyUser@$PRINCIPAL_DOMAIN" )
  conf.setIfMissing("spark.kerberos.keytab", s"$KEYTAB_DIR/$proxyUser.keytab")
}
```

**Comment by yaooqinn** [comment]
> Is it possible to first support the keytab to be stored locally on the kyuubi server.
> like:
> 
> ```
> if(USE_LOCAL_KEYTAB) {
>   conf.setIfMissing("spark.kerberos.principal", s"$proxyUser@$PRINCIPAL_DOMAIN" )
>   conf.setIfMissing("spark.kerberos.keytab", s"$KEYTAB_DIR/$proxyUser.keytab")
> }
> ```

Looks the same if users specify the local keytab in their session？

**Comment by wForget** [comment]
> > Is it possible to first support the keytab to be stored locally on the kyuubi server.
> > like:
> > ```
> > if(USE_LOCAL_KEYTAB) {
> >   conf.setIfMissing("spark.kerberos.principal", s"$proxyUser@$PRINCIPAL_DOMAIN" )
> >   conf.setIfMissing("spark.kerberos.keytab", s"$KEYTAB_DIR/$proxyUser.keytab")
> > }
> > ```
> 
> Looks the same if users specify the local keytab in their session？

Is it more convenient for users to use Kyuubi server to manage keytab files?

**Comment by turboFei** [comment]
> Is it more convenient for users to use Kyuubi server to manage keytab files?

It seems that,
1. kyuubi server should have a server keytab
2. for the engine keytab file specified by user, kyuubi server need have permission to localize this keytab file or impersonate to user account to localize it

**Comment by wForget** [comment]
> It seems that,
> 
> 1. kyuubi server should hold a server keytab
> 2. for the engine keytab file specified by user, kyuubi server need have permission to localize this keytab file(from remote filesystem like hdfs) or impersonate to user account to localize it
> 3. kyuubi server should not manage the users' keytab files locally, it is not secure and also user does not have permission to deploy the keytab file on server node

Thank you for your reply. For me, I still think this is a more convenient way, and I look forward to a more perfect way.

**Comment by yaooqinn** [comment]
`Localize spark.kerberos.keytab from remote FileSystem` means `bothj remote and local` will be supported

**Comment by yaooqinn** [comment]
> kyuubi server should not manage the users' keytab files locally

with a proper ACL, I think it can be safe

**Comment by yaooqinn** [comment]
@wForget Are you interested in working on this?

**Comment by wForget** [comment]
> @wForget Are you interested in working on this?

Yes, the company currently uses version 0.7.0 of kyuubi and hopes to upgrade to the new version of kyuubi.

It seems necessary to configure `spark.kerberos.keytab` for long-running spark applications.

**Comment by yaooqinn** [comment]
thanks @wForget
